lib/rules/jsx-no-literals.js

```diff
@@ -44,6 +44,7 @@ module.exports = {
 
     function getValidation(node) {
       const standard = !/^[\s]+$/.test(node.value) &&
+          typeof node.value === 'string' &&
```

Comment:
it looks like the `Literal` visitor doesn't hit the node when it's an object or function.